### PR TITLE
changes based on feedback for query reviewed data

### PR DESCRIPTION
**Pull request by JiaoJiao123**
It is about some feedback from Sarah. Mostly about adding 3 extra columns to gene type evidence.

### CODE REVIEW

File: web/yo/app/scripts/controllers/reportgenerator.js [hunk 1]
```diff
@@ -295,14 +295,55 @@ angular.module('oncokbApp')
 
                 DatabaseConnector.getReviewedData($scope.reviewedData[$scope.evidenceType].evidenceTypes).then(function(response) {
                     if ($scope.evidenceType === 'geneType') {
-                        _.each(response, function(item) {
-                            $scope.reviewedData.geneType.body.push({
-                                gene: item.hugoSymbol,
-                                oncogene: item.oncogene,
-                                tsg: item.tsg
+                        var variantLookupBody = _.map(response, function(item) {
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/reportgenerator.js]:
```
_.map(_.uniq(_.pluck(response, 'hugoSymbol')), function(hugoSymbol){
    return {
        hugoSymbol: hugoSymbol
    }
})
```

File: web/yo/app/scripts/controllers/reportgenerator.js [hunk 2]
```diff
@@ -295,14 +295,55 @@ angular.module('oncokbApp')
...
+                                    if (item.alteration === 'Amplification') {
+                                        tempObj.amplification = true;
+                                    }
+                                });
+                                if (!_.isEmpty(tempObj)) {
+                                    $scope.reviewedData.geneType.body.push(tempObj);
+                                }
                             });
+                            _.each(response, function(item) {
+                                if (geneWithVariants.indexOf(item.hugoSymbol) === -1) {
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/reportgenerator.js]: If this is the only place using `geneWithVariants`, suggest changing to object implementation for better performance. And this should be used for similar purpose across the project.

Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/reportgenerator.js]: Good point!

File: web/yo/app/scripts/controllers/reportgenerator.js (hunk 1 above)
Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/reportgenerator.js]: _.map can take object directly and access as key-value pairs, it is unnecessary to form an array first

Comment (zhx828) [reply web/yo/app/scripts/controllers/reportgenerator.js]: The purpose is getting the unique hugoSymbol

Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/reportgenerator.js]: I thought there is no duplication in the genes list returned by the genes API. You think there might be?

Comment (zhx828) [reply web/yo/app/scripts/controllers/reportgenerator.js]: ... oh ok. You nested genes call in here as well. If that's the case, no. There is no dup.

**Pull request merged**